src/create-glamorous.js

```diff
@@ -149,6 +149,13 @@ function createGlamorous(splitProps) {
         },
       })
 
+      function withComponent(newComp, options) {
+        return glamorous(newComp, {
+          forwardProps: GlamorousComponent.forwardProps,
+          ...options,
```

Comment:
That was interesting -- summarized: https://twitter.com/atticoos/status/872881844563447812